src/beast/beast/asio/streambuf.h

```diff
@@ -442,6 +442,7 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
                     ++pos;
                     break;
                 }
+                out_end_ = pos->size();
```

Comment:
I think so. reasoning:

When n <= avail, we set on line 452. When n > avail, we start consuming bufs. If we don't hit the end of the list, we set on line 440. If we do hit the end of the list, and n > 0, we'll allocate bufs and set on line 475

The only case I see that isn't handled is when we hit the end of the list and n is exactly 0.